[security][draft] getting rid of a recoverable panic

mimoo opened the pull request:
This should not panic as we can recover from this error: do nothing.

Sure, it might indicate some deeper problem, but we do not need to take down our validator because of this. This way we can continue to vote in subsequent round until the issue is fixed.

**draft:** I just need to understand how to properly log the error.

dmitri-perelman commented on consensus/src/chained_bft/block_storage/pending_votes.rs:
```diff
@@ -105,7 +105,7 @@ impl PendingVotes {
                 VoteReceptionResult::VoteAdded(voting_power)
             }
-            _ => panic!("Unexpected verification error, vote = {}", vote),
+            Err(_) => VoteReceptionResult::ErrorAddingVote,
```
I actually don't understand why we ever thought it's a good idea to panic in this case. Thanks for fixing this one!

davidiw replied:
Somehow this got blamed on me. I think at that point in time, the context was that we had some really strange error and while we might be able to make some progress, we'd likely be in a wonky state. Arguably we should build better log monitoring so that if we do get in that wonky state, we are alerted and perhaps our deployment framework restarts the process if it is bad enough.

dmitri-perelman commented on consensus/src/chained_bft/block_storage/mod.rs:
```diff
@@ -30,6 +30,8 @@ pub enum VoteReceptionResult {
     /// The vote completes a new TimeoutCertificate
     NewTimeoutCertificate(Arc<TimeoutCertificate>),
+    /// There might be some issues adding a vote
+    ErrorAddingVote,
```
Can we return a `VerifyError` here? (this way we'd not lose the information about the verification problem)

mimoo replied:
I was thinking that we could log the error before returning this, but if we log the error at the caller then this makes sense.

I'm still unsure on how to log this error  though :D

opsguy commented:
These are our rough guidelines:
critical: The process is unlikely to be able to continue operating, most of the time a log at this level should be followed by a crash. Examples: unable to load config at startup, unable to read/write to storage.
error: An unexpected situation has occurred which may prevent the overall system from functioning as intended. These are usually caused by code bugs or system interactions. These should never occur as the result of end user or network interactions. 

This sounds like error level to me. We don't have anything outside yet specifically to parse these and act, but just getting the level right is the most important piece right now.

mimoo commented:
Alright! addressed @dmitri-perelman comment and added an `error!("FIXME: ...")` log.

It would be good to document a way to prefix errors perhaps? 
@opsguy

mimoo commented:
@bors-libra r=dmitri-perelman

bors-libra commented:
:pushpin: Commit 482035f has been approved by `dmitri-perelman`

<!-- @bors-libra r=dmitri-perelman 482035f535e314ce1bdea1695cf11b53c90a5f97 -->

bors-libra commented:
:hourglass: Testing commit 482035f535e314ce1bdea1695cf11b53c90a5f97 with merge e1487fe918fbff132610e05e30858fc07283cc5a...

bors-libra commented:
:sunny: Test successful - [checks-circle_commit_workflow](https://circleci.com/workflow-run/1f5b7f0a-fc27-497a-85ee-6ed83a9f5a37?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)
Approved by: dmitri-perelman
Pushing e1487fe918fbff132610e05e30858fc07283cc5a to master...
